packages/snaps-execution-environments/src/common/endowments/network.ts

```diff
@@ -93,6 +93,14 @@ class ResponseWrapper implements Response {
  * function, as well as a teardown function.
  */
 const createNetwork = () => {
+  // Request, Headers and Response are the endowments injected alongside fetch
+  // only when 'endowment:network-access' permission is requested,
+  // therefore these are hardened as part of fetch dependency injection within its factory.
+  // These endowments are not (and should never be) available by default.
+  harden(Request);
```

Comment:
Can we return them as part of the endowment factory return value? Similar to how fetch is returned

Reply:
Yes, refactored.